octobot_trading/exchanges/parser/orders_parser.py

```diff
@@ -0,0 +1,730 @@
+            not_found_val=missing_side_value,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_SIDE,
+        )
+
+    def _found_side(self, raw_side):
+        return TradeOrderSide(self.TRADE_ORDER_SIDE_MAP[raw_side.lower()]).value
+
+    def _parse_price(self, missing_price_value):
+        self._try_to_find_and_set_decimal(
+            OrderCols.PRICE.value,
+            self.PRICE_KEYS,
+            not_found_val=missing_price_value,
+            enable_log=False if missing_price_value else True,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_PRICE,
+        )
+
+    def _parse_filled_price(self):
+        self._try_to_find_and_set_decimal(
+            OrderCols.FILLED_PRICE.value,
+            self.FILLED_PRICE_KEYS,
+            not_found_method=self.missing_filled_price,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_FILLED_PRICE,
+        )
+
+    def missing_filled_price(self, _):
+        if status := self.formatted_record.get(OrderCols.STATUS.value):
+            if (
+                status == OrderStatus.FILLED.value
+                or status == OrderStatus.CLOSED.value
+                or status == OrderStatus.PARTIALLY_FILLED.value
+            ) and (price := self.formatted_record.get(OrderCols.PRICE.value)):
+                return price
+            if (
+                status == OrderStatus.CANCELED.value
+                or status == OrderStatus.OPEN.value
+                or status == OrderStatus.PENDING_CANCEL.value
+                or status == OrderStatus.EXPIRED.value
+                or status == OrderStatus.PENDING_CREATION.value
+                or status == OrderStatus.REJECTED.value
+            ):
+                return 0  # to check - should we set it to None?
+        self._log_missing(
+            OrderCols.FILLED_PRICE.value,
+            f"key: {self.FILLED_PRICE_KEYS}, "
+            f"and using status ({status or 'no status'})",
+        )
+
+    def _parse_amount(self, missing_quantity_value):
+        self._try_to_find_and_set_decimal(
+            OrderCols.AMOUNT.value,
+            self.AMOUNT_KEYS,
+            not_found_val=missing_quantity_value,
+            enable_log=False if missing_quantity_value else True,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_AMOUNT,
+        )
+
+    def _parse_cost(self):
+        self._try_to_find_and_set_decimal(
+            OrderCols.COST.value,
+            self.COST_KEYS,
+            not_found_method=self.missing_cost,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_COST,
+        )
+
+    def missing_cost(self, _):
+        # check and should it be with fees?
+        filled_price = None
+        if (
+            filled_quantity := self.formatted_record.get(OrderCols.FILLED_AMOUNT.value)
+        ) is not None and (
+            filled_price := self.formatted_record.get(OrderCols.FILLED_PRICE.value)
+        ):
+            return filled_quantity * filled_price
```

Comment:
I'm not sure we should do that as it's not always true. `filled_quantity * filled_price` is valid for spot but for futures it's different. Maybe we shouldn't have a fallback func here if we don't know in which case we are.

Reply:
oh right, i changed it, so it'll now only recalculate for spot.
Or do you think we should make it optional for spot?